graphics/InsightToolkit4/Portfile

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+PortGroup           active_variants 1.1
+PortGroup           conflicts_build 1.0
+
+name                InsightToolkit4
+github.setup        InsightSoftwareConsortium ITK 4.13.3 v
+revision            0
+categories          graphics science devel
+license             Apache
+conflicts_build     InsightToolkit-devel
+
+maintainers         nomaintainer
+
+description         Insight Segmentation and Registration Toolkit (ITK)
+
+long_description    {*}${description} is an open-source, cross-platform system that \
+                    provides developers with an extensive suite of software tools \
+                    for image analysis. Developed through extreme programming \
+                    methodologies, ITK builds on a proven, spatially-oriented \
+                    architecture for processing, segmentation, and registration \
+                    of scientific images in two, three, or more dimensions.
+
+homepage            https://itk.org
+
+distname            InsightToolkit-${version}
```

Comment:
Particularly when you have devel and non-devel ports, the distfiles should be mirrored in the same location. This ensures there's no duplication (and wasted space) on various mirrors.

So you'll want to set `dist_subdir` to a common name across these ports, such as `dist_subdir InsightToolkit`.

Reply:
To illustrate what I mean, take a look at what's being mirrored for the various versions:

* https://distfiles.macports.org/InsightToolkit

* https://distfiles.macports.org/InsightToolkit312

* https://distfiles.macports.org/InsightToolkit314

There is significant duplication, which would be eliminated with use of `dist_subdir`:

BrainPart1.tgz 2018-05-04 03:36 5.6M
BrainPart1Rotated10Translated15.tgz 2018-05-04 03:36 4.9M
BrainPart1Rotated20Translated20.tgz 2018-05-04 03:36 4.3M
BrainPart2.tgz 2018-05-04 03:37 5.3M
BrainPart2Rotated10Translated15.tgz 2018-05-04 03:37 4.6M
BrainPart2Rotated20Translated20.tgz 2018-05-04 03:37 4.1M
BrainPart3.tgz 2018-05-04 03:37 5.8M
BrainPart3Rotated10Translated15.tgz 2018-05-04 03:37 5.1M
BrainPart3Rotated20Translated20.tgz 2018-05-04 03:37 4.5M
CableSwig-ITK-3.12.0.tar.gz 2015-08-29 19:52 18M
DoxygenInsightToolkit-3.12.0.tar.gz 2015-08-29 19:54 102M
InsightToolkit-3.12.0.tar.gz 2015-08-29 19:52 30M

Reply:
Does this mean I should change setting `distname` to set `dist_subdir`? Or are both needed? I suspect the latter is the case, but before I push nonsense...

Reply:
> Does this mean I should change setting `distname` to set `dist_subdir`? Or are both needed? I suspect the latter is the case, but before I push nonsense... 

No, you still need `distname` too, and that doesn't change.

Reply:
> > Does this mean I should change setting `distname` to set `dist_subdir`? Or are both needed? I suspect the latter is the case, but before I push nonsense...
> 
> No, you still need `distname` too, and that doesn't change.

Thought so. Thanks, now I understand the purpose with `dist_subdir`.